consensus/enclave/tests/enclave_api_tests.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use aes_gcm::Aes256Gcm;
+use mc_attest_ake::{AuthResponseInput, ClientInitiate, Start, Transition};
+use mc_attest_api::attest::{AuthMessage, Message};
+use mc_attest_net::{Client, RaClient};
+use mc_common::{
+    logger::{test_with_logger, Logger},
+    ResponderId,
+};
+use mc_consensus_enclave::{ConsensusEnclave, ConsensusServiceSgxEnclave, ENCLAVE_FILE};
+use mc_consensus_enclave_api::{BlockchainConfig, ClientProposeTxRequestV2, FeeMap};
+use mc_crypto_keys::X25519;
+use mc_crypto_rand::McRng;
+use mc_fog_test_infra::get_enclave_path;
+use mc_ledger_db::{
+    test_utils::{create_ledger, create_transaction, initialize_ledger},
+    Ledger,
+};
+use mc_sgx_report_cache_untrusted::ReportCache;
+use mc_transaction_core::{AccountKey, BlockVersion};
+use mc_util_metrics::IntGauge;
+use mc_util_serial::encode;
+use sha2::Sha512;
+use std::str::FromStr;
+
+lazy_static::lazy_static! {
+    pub static ref DUMMY_INT_GAUGE: IntGauge = IntGauge::new("foo".to_string(), "bar".to_string()).unwrap();
+}
+
+/// Test that we can exercise client_tx_propose_v2 and that it passes and fails
+/// as expected.
+#[test_with_logger]
+fn consensus_enclave_client_tx_propose_v2(logger: Logger) {
```

Comment:
I was able to create a proper test of the consensus enclave client_propose_tx_v2 this way